sql/core/src/test/scala/org/apache/spark/sql/execution/SelectedFieldSuite.scala

```diff
@@ -27,309 +27,373 @@ import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
-          StructField("subfield1", IntegerType) ::
-          StructField("subfield2", ArrayType(IntegerType, containsNull = false)) :: Nil)), valueContainsNull = false)) ::
-        StructField("field9", MapType(StringType, IntegerType, valueContainsNull = false)) :: Nil)) ::
-     StructField("col3", ArrayType(StructType(
-       StructField("field1", StructType(
-         StructField("subfield1", IntegerType, nullable = false) ::
-         StructField("subfield2", IntegerType) :: Nil)) ::
-       StructField("field2", MapType(StringType, IntegerType, valueContainsNull = false)) :: Nil), containsNull = false), nullable = false) ::
-     StructField("col4", MapType(StringType, StructType(
-       StructField("field1", StructType(
-         StructField("subfield1", IntegerType, nullable = false) ::
-         StructField("subfield2", IntegerType) :: Nil)) ::
-       StructField("field2", MapType(StringType, IntegerType, valueContainsNull = false)) :: Nil), valueContainsNull = false), nullable = false) ::
-     StructField("col5", ArrayType(MapType(StringType, StructType(
-       StructField("field1", StructType(
-         StructField("subfield1", IntegerType) ::
-         StructField("subfield2", IntegerType) :: Nil)) :: Nil), valueContainsNull = false))) ::
-     StructField("col6", MapType(StringType, ArrayType(StructType(
-       StructField("field1", StructType(
-         StructField("subfield1", IntegerType) ::
-         StructField("subfield2", IntegerType) :: Nil)) :: Nil), containsNull = false))) ::
-     StructField("col7", ArrayType(StructType(
-       StructField("field1", IntegerType, nullable = false) ::
-       StructField("field2", StructType(
-         StructField("subfield1", IntegerType, nullable = false) :: Nil)) ::
-       StructField("field3", ArrayType(StructType(
-         StructField("subfield1", IntegerType, nullable = false) :: Nil))) :: Nil))) ::
-     StructField("col8", ArrayType(StructType(
-       StructField("field1", ArrayType(IntegerType, containsNull = false), nullable = false) :: Nil))) :: Nil)
-
-  private val testRelation = LocalRelation(schema.toAttributes)
+        StructField("field9",
+          MapType(StringType, IntegerType, valueContainsNull = false)) :: Nil)) :: Nil)
 
   test("SelectedField should not match an attribute reference") {
-    assertResult(None)(unapplySelect("col1"))
-    assertResult(None)(unapplySelect("col1 as foo"))
-    assertResult(None)(unapplySelect("col2"))
+    val testRelation = LocalRelation(nestedComplex.toAttributes)
+    assertResult(None)(unapplySelect("col1", testRelation))
+    assertResult(None)(unapplySelect("col1 as foo", testRelation))
+    assertResult(None)(unapplySelect("col2", testRelation))
   }
 
-  info("For a relation with schema\n" + indent(schema.treeString))
+  //  |-- col1: string (nullable = false)
+  //  |-- col2: struct (nullable = true)
+  //  |    |-- field1: integer (nullable = true)
```

Comment:
I don't see `field1` in `structOfArray`. Please remove it here or add it there.

Reply:
You're correct, good catch. I've removed this one. `field1` is belongs in the previous schema.